File: B面向对象/#复习最新课件/unit2 java Class Implementation(更新部分)/java-source/2.5/TestClient.java

import  java.util.*;
import  java.io.*;

/**
 * This class tests the implementation of class <code>Client</code>.
 *
  
 */
public class TestClient  {

	/* Standard output stream */
	private static PrintWriter  stdOut =
		new  PrintWriter(System.out, true);

	/* Standard error stream */
	private static PrintWriter  stdErr =
		new  PrintWriter(System.err, true);

	/**
	 * Tests the implementation of class <code>Client</code>.
	 *
	 * @param args  not used.
	 */
	public static void main(String[] args) {

		BankAccount accountOne = new BankAccount();
		BankAccount accountTwo = new BankAccount();
		BankAccount accountThree = new BankAccount();

		accountOne.deposit(1000.0);
		accountTwo.deposit(2000.0);
		accountThree.deposit(3000.0);
		
		Client client= new Client("John Smith");

		client.addAccount(accountOne);
		client.addAccount(accountTwo);
		client.addAccount(accountThree);

		double totalBalance = 0.0;
		
		for (BankAccount account : client) {
			totalBalance += account.getBalance();
		}
		
		if (totalBalance != 6000.0) {
			stdErr.println("** Test failure");
		}
		
		stdOut.println("done");
	}
}

/*!End Snippet:file*/

File: B面向对象/#复习最新课件/unit1 Object-oriented foundation and class diagram（更新）/source/1.2.3/TestClient.cpp
/*
 * TestClient.cpp
 *
 *  Created on: 2013-10-25
 *      Author: Administrator
 */
    #include "BankAccount.h"
    #include "client.h"
    #include <iostream>
    #include <vector>
    using namespace std;

	int main() {
		BankAccount accountOne;
		BankAccount accountTwo;
		BankAccount accountThree;

		accountOne.deposit(1000.0);
		accountTwo.deposit(2000.0);
		accountThree.deposit(3000.0);

		Client client("John Smith");

		client.addAccount(accountOne);
		client.addAccount(accountTwo);
		client.addAccount(accountThree);

		double totalBalance = 0.0;

		/*
		for (vector<BankAccount>::iterator iter = client.iteratorBegin(); iter!=client.iteratorEnd(); iter++) {
			totalBalance += iter->getBalance();
			//cout<<"debugging"<<iter->getBalance()<<endl;
		}*/

		for (vector<BankAccount>::size_type index = 0; index!=client.getNumberOfAccounts(); index++) {
					totalBalance += client[index].getBalance();
					//cout<<"debugging"<<iter->getBalance()<<endl;
		}
		if (totalBalance != 6000.0) {
			cout<<totalBalance<<"** Test failure"<<endl;
		}
		    cout<<"done"<<endl;
		return 0;
	}


/*!End Snippet:file*/
